miden-lib/src/notes/mod.rs

```diff
@@ -1,99 +1,90 @@
 use miden_objects::{
-    accounts::AccountId,
-    assembly::ProgramAst,
-    assets::Asset,
-    notes::{Note, NoteScript},
-    utils::{collections::Vec, vec},
-    Digest, Felt, Hasher, NoteError, Word, ZERO,
+    accounts::AccountId, assets::Asset, crypto::rand::FeltRng, notes::Note,
+    utils::collections::Vec, Felt, NoteError, Word, ZERO,
 };
 
-use super::transaction::TransactionKernel;
+use self::utils::build_note_script;
+
+pub mod utils;
 
 // STANDARDIZED SCRIPTS
 // ================================================================================================
 
-pub enum Script {
-    P2ID { target: AccountId },
-    P2IDR { target: AccountId, recall_height: u32 },
-    SWAP { asset: Asset, serial_num: Word },
-}
-
-/// Users can create notes with a standard script. Atm we provide three standard scripts:
-/// 1. P2ID - pay to id.
-/// 2. P2IDR - pay to id with recall after a certain block height.
-/// 3. SWAP - swap of assets between two accounts.
-pub fn create_note(
-    script: Script,
-    assets: Vec<Asset>,
+/// Generates a P2ID note - pay to id note.
+/// This script enables the transfer of assets from one account `sender` to another account`target`
```

Comment:
I would write this like so:
```
/// Generates a P2ID note - pay to id note.
///
/// This script enables the transfer of assets from the `sender` account to the `target` account.
/// The passed-in `rng` is used to generate a serial number for the note.
///
/// # Errors
/// Returns an error if deserialization or compilation of the `P2ID` script fails.
```

Putting a blank line between the intro sentence and the rest of the comment makes the comments look better when rustdoc formats them.

Could you apply similar changes to the other functions here?

Reply:
Could we also say that they would return an error if the note creation fails?